zerver/tests/test_embedded_bot_system.py

```diff
@@ -52,6 +53,31 @@ def test_stream_message_to_embedded_bot(self) -> None:
         display_recipient = get_display_recipient(last_message.recipient)
         self.assertEqual(display_recipient, "Denmark")
 
+    def test_embedded_bot_react_to_stream_message(self) -> None:
+        assert self.bot_profile is not None
```

Comment:
A test for failing reaction will help, just to see we get the appropriate Exception raised.

Reply:
Agree, I have added new test cases to support that.